### PR TITLE
deltas: Fix leak of matches

### PR DESCRIPTION
**Pull request by cgwalters**
Found by ASAN.

### CODE REVIEW

File: src/libostree/ostree-repo-static-delta-compilation.c [hunk 1]
```diff
@@ -636,7 +636,7 @@ try_content_rollsum (OstreeRepo                       *repo,
   if (!get_unpacked_unlinked_content (repo, to, &tmp_to, cancellable, error))
     return FALSE;
 
-  OstreeRollsumMatches *matches = _ostree_compute_rollsum_matches (tmp_from, tmp_to);
+  g_autofree OstreeRollsumMatches *matches = _ostree_compute_rollsum_matches (tmp_from, tmp_to);
```
Comment (jlebon) [inline src/libostree/ostree-repo-static-delta-compilation.c:639]: Shouldn't this be

```suggestion
  g_autoptr(OstreeRollsumMatches) matches = _ostree_compute_rollsum_matches (tmp_from, tmp_to);
```

?

Comment (cgwalters) [reply src/libostree/ostree-repo-static-delta-compilation.c]: Yep done

**Comment by cgwalters** [comment]
/override continuous-integration/jenkins/pr-merge 
Nothing in that CI runs deltas right now

**Pull request merged**